rhessys/util/readtag_worldfile.c

```diff
@@ -89,8 +89,9 @@ param *readtag_worldfile(int *paramCnt, FILE *file,char *key){
 	    if(strcmp(strbuf2,"n_basestations")==0){
 	      break;
 	    }
+	    
```

Comment:
You could try the hash table in the [POSIX standard](http://pubs.opengroup.org/onlinepubs/009695399/functions/hcreate.html), which is also described in the [GNU](http://www.gnu.org/software/libc/manual/html_node/Hash-Search-Function.html) docs.